wpimath/src/main/native/cpp/util/PIDConstants.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "frc/util/PIDConstants.h"
+
+#include "wpimath/MathShared.h"
+
+using namespace frc;
+
+PIDConstants::PIDConstants(double Kp, double Ki, double Kd)
+    : Kp{Kp}, Ki{Ki}, Kd{Kd} {
+  bool invalidGains = false;
+  if (Kp < 0.0) {
+    wpi::math::MathSharedStore::ReportError(
+        "Kp must be a non-negative number, got {}!", Kp);
+    invalidGains = true;
+  }
+  if (Ki < 0.0) {
+    wpi::math::MathSharedStore::ReportError(
+        "Ki must be a non-negative number, got {}!", Ki);
+    invalidGains = true;
+  }
+  if (Kd < 0.0) {
+    wpi::math::MathSharedStore::ReportError(
+        "Kd must be a non-negative number, got {}!", Kd);
+    invalidGains = true;
+  }
+  if (invalidGains) {
+    Kp = 0.0;
+    Ki = 0.0;
+    Kd = 0.0;
+    wpi::math::MathSharedStore::ReportWarning("PID gains defaulted to 0.");
+  }
```

Comment:
This cannot work with the values being marked const.

Reply:
If its being evaluated at compile time isn't that ok?

Reply:
It can only be initialized in the intializer list if it is marked const, even if evaluated at compile time(which constexpr does not necessarily enforce).